File: 프로그래머스/lv3/12987. 숫자 게임/숫자 게임.py
import heapq
def solution(A, B):
    answer = 0
    idx=0
    q=[]
    A.sort(reverse=True)
    B.sort(reverse=True)
    for cur in A:
        while idx<len(B) and B[idx]>cur:
            heapq.heappush(q,B[idx])
            idx+=1
        if q and q[0]>cur:
            answer+=1
            heapq.heappop(q)
    return answer

File: 프로그래머스/lv3/12987. 숫자 게임/숫자 게임.java
import java.util.*;
class Solution {
    public int solution(int[] A, int[] B) {
        int answer = 0;
        Arrays.sort(A);
        Arrays.sort(B);
        int num=0;
        for (int i = 0; i < B.length; i++) {
            num=Arrays.binarySearch(B,num,B.length, A[i]+1);
            if (num<0)
                num*=-1;
            if (num>B.length)break;
            answer+=1;
        }
        return answer;
    }
}